core/flow/rule_manager.go

```diff
@@ -270,30 +431,35 @@ func IsValidRule(rule *Rule) error {
 	if rule.Resource == "" {
 		return errors.New("empty resource name")
 	}
-	if rule.Count < 0 {
+	if rule.Threshold < 0 {
 		return errors.New("negative threshold")
 	}
-	if rule.MetricType < 0 {
-		return errors.New("invalid metric type")
+	if int32(rule.TokenCalculateStrategy) < 0 {
+		return errors.New("invalid token calculate strategy")
 	}
-	if rule.RelationStrategy < 0 {
-		return errors.New("invalid relation strategy")
+	if int32(rule.ControlBehavior) < 0 {
+		return errors.New("invalid control behavior")
 	}
-	if rule.TokenCalculateStrategy < 0 || rule.ControlBehavior < 0 {
-		return errors.New("invalid control strategy")
+	if !(rule.RelationStrategy >= CurrentResource && rule.RelationStrategy <= AssociatedResource) {
+		return errors.New("invalid relation strategy")
 	}
-
 	if rule.RelationStrategy == AssociatedResource && rule.RefResource == "" {
-		return errors.New("Bad flow rule: invalid control behavior")
+		return errors.New("Bad flow rule: invalid relation strategy")
 	}
-
 	if rule.TokenCalculateStrategy == WarmUp {
 		if rule.WarmUpPeriodSec <= 0 {
-			return errors.New("invalid warmUpPeriodSec")
+			return errors.New("invalid WarmUpPeriodSec")
 		}
 		if rule.WarmUpColdFactor == 1 {
 			return errors.New("WarmUpColdFactor must be great than 1")
 		}
 	}
+	if rule.ControlBehavior == Throttling && rule.MaxQueueingTimeMs == 0 {
+		return errors.New("invalid MaxQueueingTimeMs")
+	}
+	if !((rule.StatIntervalInMs == 0 && rule.StatSampleCount == 0) ||
```

Comment:
Maybe we could split the large condition to several checkings and make the message clearer.